packages/file_selector/file_selector_android/android/src/main/java/io/flutter/plugins/file_selector/ActivityStateHelper.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.file_selector;
+
+import android.app.Activity;
+import android.app.Application;
+import androidx.annotation.VisibleForTesting;
+import androidx.lifecycle.Lifecycle;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.embedding.engine.plugins.lifecycle.FlutterLifecycleAdapter;
+
+/**
+ * Move all activity-lifetime-bound states into this helper object, so that {@code setup} and {@code
```

Comment:
"Owns all activity-lifetime-bound state, to simplify lifetime state handling for clients."

(The current comment describes a point-in-time refactor—which never even happened in this plugin since it's newer than the change in the code this is copied from—rather than what the class actually does.)